src/utils/stats_alloc/src/task_local.rs

```diff
@@ -0,0 +1,184 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// FIXME: This is a false-positive clippy test, remove this while bumping toolchain.
+// https://github.com/tokio-rs/tokio/issues/4836
+// https://github.com/rust-lang/rust-clippy/issues/8493
+#![expect(clippy::declare_interior_mutable_const)]
+
+use std::alloc::{GlobalAlloc, Layout, System};
+use std::future::Future;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::time::Duration;
+
+use tokio::task_local;
+
+#[repr(transparent)]
+#[derive(Clone, Copy)]
+struct TaskLocalBytesAllocated(Option<&'static AtomicUsize>);
```

Comment:
If we're going to release it instead of leaking it, maybe using a `NonNull` pointer here will be more suitable?